bootstrap/hook.go

```diff
@@ -50,10 +50,14 @@ func newHookScriptWrapper(hookPath string) (*hookScriptWrapper, error) {
 	var err error
 	var scriptFileName string = `buildkite-agent-bootstrap-hook-runner`
 	var isBashHook bool
+	var isPwshHook bool
```

Comment:
I wonder if this should change to an enum (or some Go equivalent) rather than two booleans that could contradict each other if both are true.

Reply:
Also maybe worth copying the `var isWindows = runtime.GOOS == "windows"` from `shell.go` to make the conditionals below a bit simpler to read.

Reply:
thought about how this could be accomplished in Go but haven't fully fledged out the implementation yet. I can revisit this in the future